chore: release notes for v1.49 js (#33565)
microsoft/playwright: docs/src/release-notes-js.md

## Version 1.49

### Aria snapshots

> New assertion [`method: LocatorAssertions.toMatchAriaSnapshot`] provides a way to verify page structure by comparing it with an expected YAML representation of the accessibility tree.

Reviewer: nit: making this a little more succinct.

```suggestion
The new assertion [`method: LocatorAssertions.toMatchAriaSnapshot`] verifies page structure by comparing to an expected accessibility tree, represented as YAML.
```

  - banner:
    - heading /Playwright enables reliable/ [level=1]
    - link "Get started"
    - link "Star microsoft/playwright on GitHub"
  - main:
    - img "Browsers (Chromium, Firefox, WebKit)"
    - heading "Any browser • Any platform • One API"
`);
```

You can generate this assertion with [Test Generator](./codegen) and update the expected snapshot with `--update-snapshots` command line flag.

Learn more in the [aria snapshots guide](./aria-snapshots).

### Test runner

- New option [`property: TestConfig.tsconfig`] allows to specify a single `tsconfig` to be used for all tests.
- New method [`method: Test.fail.only`] to focus on a failing test.
- Options [`property: TestConfig.globalSetup`] and [`property: TestConfig.globalTeardown`] now support multiple setups/teardowns.
- New value `'on-first-failure'` for [`property: TestOptions.screenshot`].
- Added "previous" and "next" buttons to the HTML report to quickly switch between test cases.
- New properties [`property: TestInfoError.cause`] and [`property: TestError.cause`] mirroring [`Error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

### Miscellaneous

> - `<canvas>` elements inside a snapshot now draw a preview.

Reviewer 2: Lets wait with landing until https://github.com/microsoft/playwright/issues/33562#issuecomment-2471874482 is resolved?

Reviewer: Agree!

Author: We'll update if needed.

- New method [`method: Tracing.group`] to visually group actions in the trace.
- Playwright docker images switched from Node.js v20 to Node.js v22 LTS.

### Breaking changes

> - There will be no more updates for WebKit on Ubuntu 20.04. We recommend updating Ubuntu to a later version.

Reviewer 2: Lets mention Debian 11 here as well.
